packages/@webex/plugin-cc/src/services/task/TaskManager.ts

```diff
@@ -51,18 +58,30 @@ export default class TaskManager extends EventEmitter {
       if (payload.data) {
         switch (payload.data.type) {
           case CC_EVENTS.AGENT_CONTACT_RESERVED:
-            this.task = new Task(this.contact, this.webCallingService, payload.data);
+            this.currentTask = new Task(this.contact, this.webCallingService, payload.data);
 
-            this.taskCollection[payload.data.interactionId] = this.task;
+            this.taskCollection[payload.data.interactionId] = this.currentTask;
             if (this.webCallingService.loginOption !== LoginOption.BROWSER) {
-              this.emit(TASK_EVENTS.TASK_INCOMING, this.task);
+              this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
             } else if (this.call) {
-              this.emit(TASK_EVENTS.TASK_INCOMING, this.task);
+              this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
+              // this.currentTask.webCallMap[this.currentTask.data.interactionId] = this.call.getCallId();
             }
             break;
           case CC_EVENTS.AGENT_CONTACT_ASSIGNED:
-            this.task = this.task.updateTaskData(payload.data);
-            this.emit(TASK_EVENTS.TASK_ASSIGNED, this.task);
+            this.currentTask = this.currentTask.updateTaskData(payload.data);
+            this.emit(TASK_EVENTS.TASK_ASSIGNED, this.currentTask);
+            break;
+          case CC_EVENTS.CONTACT_ENDED:
+            this.emit(TASK_EVENTS.TASK_UNASSIGNED, this.currentTask);
+            if (this.webCallingService.loginOption === LoginOption.BROWSER) {
+              this.currentTask.unregisterWebCallListeners();
+              this.webCallingService.unregisterCallListeners();
+            }
+            break;
+          case CC_EVENTS.AGENT_WRAPUP:
```

Comment:
Need this in case user on other end drops off... there is an issue with the reading of events. 
Will sync and look into this issue.